File: ruby/difference-of-squares/difference_of_squares.rb
class Squares
  def initialize(number)
    @number = number
  end

  def square_of_sum
    range.map { |num| num }.sum ** 2
  end

  def sum_of_squares
    range.map { |num| num ** 2 }.sum
  end

  def difference
    square_of_sum - sum_of_squares
  end

  private
    def range
      (1..@number).to_a
    end
end

module BookKeeping
  VERSION = 4
end

File: python/difference-of-squares/difference_of_squares.py
def square_of_sum(count):
    return reduce_array(range(1, count + 1)) ** 2

def sum_of_squares(count):
    squares = list(map(lambda x: x ** 2, range(1, count + 1)))
    return reduce_array(squares)

def reduce_array(array):
    return reduce(lambda x, y: x + y, array)

def difference(count):
    return square_of_sum(count) - sum_of_squares(count)
